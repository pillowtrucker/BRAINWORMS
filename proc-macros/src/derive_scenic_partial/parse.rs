use proc_macro2::TokenStream;
use proc_macro_error::abort;
use syn::parse::{Parse, ParseStream};
use syn::{parse2, Result};

pub struct Ast {}

impl Parse for Ast {
    fn parse(_stream: ParseStream) -> Result<Self> {
        Ok(Ast {})
    }
}

pub fn parse(ts: TokenStream) -> Ast {
    match parse2::<Ast>(ts) {
        Ok(ast) => ast,
        Err(e) => {
            abort!(e.span(), e)
        }
    }
}

#[cfg(test)]
mod tests {
    use quote::quote;

    use super::*;
    #[test]
    fn valid_syntax() {
        parse(quote!());
    }
}
