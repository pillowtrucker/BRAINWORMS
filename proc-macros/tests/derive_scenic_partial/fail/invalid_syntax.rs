use proc_macros::derive_scenic_partial;

fn main() {
    derive_scenic_partial!(meow);
}
