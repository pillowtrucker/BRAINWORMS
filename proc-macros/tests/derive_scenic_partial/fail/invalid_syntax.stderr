error: unexpected token
 --> tests/fail/invalid_syntax.rs:4:12
  |
4 |     derive_scenic_partial!(meow);
  |                     ^^^^
